redis/conn.go

```diff
@@ -351,32 +352,50 @@ func (c *conn) writeCommand(cmd string, args []interface{}) (err error) {
 		if err != nil {
 			break
 		}
-		switch arg := arg.(type) {
-		case string:
-			err = c.writeString(arg)
-		case []byte:
-			err = c.writeBytes(arg)
-		case int:
-			err = c.writeInt64(int64(arg))
-		case int64:
-			err = c.writeInt64(arg)
-		case float64:
-			err = c.writeFloat64(arg)
-		case bool:
-			if arg {
-				err = c.writeString("1")
-			} else {
-				err = c.writeString("0")
-			}
-		case nil:
-			err = c.writeString("")
+		err = c.writeArg(arg)
+	}
+	return err
+}
+
+func (c *conn) writeArg(arg interface{}) (err error) {
+	switch arg := arg.(type) {
+	case string:
+		err = c.writeString(arg)
+	case []byte:
+		err = c.writeBytes(arg)
+	case int:
+		err = c.writeInt64(int64(arg))
+	case int64:
+		err = c.writeInt64(arg)
+	case float64:
+		err = c.writeFloat64(arg)
+	case bool:
+		if arg {
+			err = c.writeString("1")
+		} else {
+			err = c.writeString("0")
+		}
+	case nil:
+		err = c.writeString("")
+	default:
+		v := reflect.ValueOf(arg)
+		switch v.Kind() {
```

Comment:
I'm unsure of the potential performance penalty to `reflect.ValueOf()`... if it's inexpensive, it may make more sense to _always_ do this, and use the `v.Kind()` switch on line 361 instead of the `arg.(type)` switch.  That would avoid the recursion and make this a little less complicated-looking.